hal/src/rtl872x/backup_ram_hal.cpp

```diff
@@ -48,17 +96,34 @@ int hal_backup_ram_init(void) {
         CHECK(hal_flash_read((uintptr_t)&platform_backup_ram_persisted_flash_start, (uint8_t*)&platform_backup_ram_all_start,
                 (size_t)&platform_backup_ram_persisted_flash_size));
     }
-    return 0;
+    return SYSTEM_ERROR_NONE;
 }
 
-int hal_backup_ram_sync(void) {
+int hal_backup_ram_sync(void* reserved) {
+    BackupRamLock lk;
     if (memcmp((void*)&platform_backup_ram_all_start,
             (void*)&platform_backup_ram_persisted_flash_start,
             (uintptr_t)&platform_backup_ram_persisted_flash_size)) {
+        // FIXME: use a more generic flag name, instead of entered_hibernate.
```

Comment:
Something like `restore_backup_ram`? That seems to convey the purpose of this flag?

Reply:
Sounds good to me.